OptimizelySDKCore/OptimizelySDKCore/OPTLYEventBuilder.m

```diff
@@ -14,114 +14,113 @@
-                                userId:(NSString *)userId
-                             eventName:(NSString *)eventName
-                             eventTags:(NSDictionary *)eventTags
-                            attributes:(NSDictionary<NSString *,NSString *> *)attributes {
-    if (!config) {
+
+-(NSDictionary *)buildConversionEventTicketForUser:(NSString *)userId
+                                             event:(OPTLYEvent *)event
+                                         decisions:(NSArray<NSDictionary *> *)decisions
+                                         eventTags:(NSDictionary *)eventTags
+                                        attributes:(NSDictionary<NSString *,NSString *> *)attributes {
+
+    if (!self.config) {
         return nil;
     }
     
-    if (!bucketer) {
-        [config.logger logMessage:OPTLYLoggerMessagesBucketerInvalid withLevel:OptimizelyLogLevelWarning];
+    if ([OPTLYEventBuilderDefault isEmptyString:userId]) {
+        [self.config.logger logMessage:OPTLYLoggerMessagesUserIdInvalid withLevel:OptimizelyLogLevelError];
         return nil;
     }
     
-    if ([userId length] == 0) {
-        [config.logger logMessage:OPTLYLoggerMessagesUserIdInvalid withLevel:OptimizelyLogLevelWarning];
+    if (!event) {
+        [self.config.logger logMessage:OPTLYLoggerMessagesNoEventForConversionEventTicket withLevel:OptimizelyLogLevelError];
         return nil;
     }
     
-    if ([eventName length] == 0) {
-        [config.logger logMessage:OPTLYLoggerMessagesEventKeyInvalid withLevel:OptimizelyLogLevelWarning];
+    if (!decisions) {
+        [self.config.logger logMessage:OPTLYLoggerMessagesNoDecisionForConversionEventTicket withLevel:OptimizelyLogLevelError];
         return nil;
     }
     
-    NSDictionary *commonParams = [self createCommonParams:config userId:userId attributes:attributes];
-    NSArray *conversionOnlyParams = [self createConversionParams:config bucketer:bucketer eventKey:eventName userId:userId eventTags:eventTags attributes:attributes];
-    if ([conversionOnlyParams count] == 0) {
-        [config.logger logMessage:OPTLYLoggerMessagesVariationIdInvalid withLevel:OptimizelyLogLevelWarning];
-        return nil;
-    }
+    NSDictionary *commonParams = [self createCommonParamsForUser:userId attributes:attributes];
+    NSArray *conversionOnlyParams = [self createConversionParamsOfEvent:event userId:userId
+                                                              decisions:decisions eventTags:eventTags
+                                                             attributes:attributes];
     NSDictionary *conversionParams = [self createImpressionOrConversionParamsWithCommonParams:commonParams conversionOrImpressionOnlyParams:conversionOnlyParams];
     
     return conversionParams;
 }
 
 
-- (NSNumber *)revenueValue:(OPTLYProjectConfig *)config value:(NSObject *)value {
+- (NSNumber *)revenueValueFromEventTags:(NSDictionary *)eventTags {
```

Comment:
can we use external alias in argument instead of FromTags in the method name.

Reply:
@thomaszurkan-optimizely I think these two methods revenueValueFromEventTags and numericValue should be moved in Utils class and this builder class should call those methods. 
Shouldnt' be a part of this class. 
What do you suggest?

Reply:
Agreed. The other SDKs should already be doing that